### PR TITLE
Adding insights credential to tower_inventory

### LINKED ISSUE #8015: Add insignts_credential paramter to tower_inventory
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY

Per PR #7963 tower_inventory is missing support for the insights_credential API parameter.

### PR DESCRIPTION
**Pull request by john-westcott-iv**
Fixed a spacing issue in documentation

Enhanced integration test to include a block to ensure cleanup of assets

##### SUMMARY

Fixes #8015 

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - Collections

##### AWX VERSION

```
awx: 14.1.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by AlanCoding** [comment]
update the completeness test??

**Comment by john-westcott-iv** [comment]
> update the completeness test??

Done, was just waiting for its merge.

File: awx_collection/tests/integration/targets/tower_inventory/tasks/main.yml [hunk 1]
```diff
@@ -1,101 +1,142 @@
...
+        organization: Default
+        insights_credential: "{{ cred_name1 }}"
+        state: present
+      register: result
+
+    - assert:
+        that:
+          - "result is changed"
+
+    - meta: end_play
```
Comment (beeankha) [inline awx_collection/tests/integration/targets/tower_inventory/tasks/main.yml]: Should this line be in here?  I had to take it out in order to locally run the integration playbook fully.

Comment (john-westcott-iv) [reply awx_collection/tests/integration/targets/tower_inventory/tasks/main.yml]: Thank you, this was a troubleshooting thing and should not be there.

File: awx_collection/test/awx/test_inventory.py [hunk 2]
```diff
@@ -3,20 +3,26 @@
...
+    # Create an insights credential
+
     result = run_module('tower_inventory', {
         'name': 'foo-inventory',
         'organization': organization.name,
         'variables': {'foo': 'bar', 'another-foo': {'barz': 'bar2'}},
+        'insights_credential': insights_credential.name,
         'state': 'present'
     }, admin_user)
+    assert not result.get('failed', False), result.get('msg', result)
```
Comment (AlanCoding) [inline awx_collection/test/awx/test_inventory.py:21]: ha, I know it's a test I wrote early on when it doesn't have the recipe to correctly assert for success

File: awx_collection/test/awx/test_completeness.py [hunk 3]
```diff
@@ -58,7 +58,6 @@
 ]
 needs_param_development = {
     'tower_host': ['instance_id'],
-    'tower_inventory': ['insights_credential'],
```
Comment (AlanCoding) [inline awx_collection/test/awx/test_completeness.py:61]: 💥

File: awx_collection/plugins/modules/tower_inventory.py [hunk 4]
```diff
@@ -48,7 +48,11 @@
       type: str
     host_filter:
       description:
-        -  The host_filter field. Only useful when C(kind=smart).
+        - The host_filter field. Only useful when C(kind=smart).
+      type: str
+    insights_credential:
+      description:
+        - Credentials to be used by hosts belonging to this inventory when accessing Red Hat Insights API.
```
Comment (AlanCoding) [inline awx_collection/plugins/modules/tower_inventory.py:55]: credentials, plural? hmm... I guess maybe that makes sense, like "data".

**Review by AlanCoding — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**